librz/util/table.c

```diff
@@ -60,16 +60,34 @@ static void __table_adjust(RzTable *t) {
 	}
 }
 
-RZ_API void rz_table_row_free(void *_row) {
+/**
+ * \brief Free function for RzVector rows in RzTable
+ *
+ * \param _row pointer to the elements of rows in RzTable
+ */
+RZ_API void rz_table_row_free(RZ_NONNULL void *_row) {
+	rz_return_if_fail(_row);
 	RzTableRow *row = _row;
-	rz_list_free(row->items);
-	free(row);
+	rz_pvector_free(row->items);
 }
 
-RZ_API void rz_table_column_free(void *_col) {
+static void rz_table_row_free_wrapper(RZ_NONNULL void *_row, void *user) {
+	rz_table_row_free(_row);
+}
+
+/**
+ * \brief Free function for RzVector cols in RzTable
+ *
+ * \param _col pointer to the elements of cols in RzTable
+ */
+RZ_API void rz_table_column_free(RZ_NONNULL void *_col) {
+	rz_return_if_fail(_col);
 	RzTableColumn *col = _col;
 	free(col->name);
-	free(col);
```

Comment:
why removing this?

Reply:
Thanks for your review. Different from RzList, RzVector owns a continuous memory. IMHO, calling `free(col)` here will not begin from the beginning of the allocated memory block. Instead, the memory will be freed when the whole vector is destroyed.